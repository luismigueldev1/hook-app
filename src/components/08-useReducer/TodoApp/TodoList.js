import React from "react";
import TodoListItem from "./TodoListItem";
import PropTypes from "prop-types";

function TodoList({ todos, handleDelete, handleToogle }) {
  return (
    <ul className="p-0 list-group ">
      {todos.map((todo) => (
        <TodoListItem
          key={todo.id}
          todo={todo}
          handleDelete={handleDelete}
          handleToogle={handleToogle}
        />
      ))}
    </ul>
  );
}

TodoList.propTypes = {
  todos: PropTypes.array.isRequired,
  handleDelete: PropTypes.func.isRequired,
  handleToogle: PropTypes.func.isRequired,
};

export default TodoList;
