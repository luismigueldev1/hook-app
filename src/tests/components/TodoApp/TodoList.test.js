import { shallow } from "enzyme";
import TodoList from "../../../components/08-useReducer/TodoApp/TodoList";
import { demoTodos } from "../../fixtures/demoTodos";

describe("Pruebas en <TodoList/>", () => {
  const handleDelete = jest.fn();
  const handleToogle = jest.fn();

  const wrapper = shallow(
    <TodoList
      todos={demoTodos}
      handleDelete={handleDelete}
      handleToogle={handleToogle}
    />
  );

  test("debe mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de tener 2 <TodoListItem/>", () => {
    expect(wrapper.find("TodoListItem").length).toBe(demoTodos.length);
    expect(wrapper.find("TodoListItem").at(0).prop("handleDelete")).toEqual(
      expect.any(Function)
    );
  });
});
