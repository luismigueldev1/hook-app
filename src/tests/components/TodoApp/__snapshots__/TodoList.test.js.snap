// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoList/> debe mostrarse correctamente 1`] = `
<ul
  className="p-0 list-group "
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToogle={[MockFunction]}
    key="1"
    todo={
      Object {
        "done": false,
        "id": 1,
        "task": "Aprender React",
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToogle={[MockFunction]}
    key="2"
    todo={
      Object {
        "done": false,
        "id": 2,
        "task": "Aprender React Native",
      }
    }
  />
</ul>
`;
